parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -101,10 +103,17 @@ impl NativeTokensMessage {
 				create_call_index,
 				set_metadata_call_index,
 			} => {
-				let owner = GlobalConsensusEthereumAccountConvertsFor::<[u8; 32]>::from_params(
-					&chain_id,
-					origin.as_fixed_bytes(),
-				);
+				let owner =
+					GlobalConsensusEthereumAccountConvertsFor::<[u8; 32]>::convert_location(
+						&MultiLocation::new(
+							2,
+							(
+								GlobalConsensus(Ethereum { chain_id }),
```

Comment:
We can reuse the network id:
```suggestion
								GlobalConsensus(network),
```